corehq/apps/case_importer/tracking/case_upload_tracker.py

```diff
@@ -13,48 +15,51 @@ def __init__(self, upload_id):
         self.upload_id = upload_id
 
     @classmethod
-    def create(cls, payload, file_extension):
-        soil_download = expose_cached_download(
-            payload, expiry=cls._expiry, file_extension=file_extension)
-        return cls(soil_download.download_id)
+    def create(cls, file_object, filename):
+        upload_id = transient_file_store.write_file(file_object, filename)
+        return cls(upload_id)
 
     @classmethod
     def get(cls, upload_id):
         return cls(upload_id)
 
     @property
     @memoized
-    def _soil_download(self):
-        return DownloadBase.get(self.upload_id)
-
-    @property
-    @memoized
     def _case_upload_record(self):
         return CaseUploadRecord.objects.get(upload_id=self.upload_id)
 
-    def get_filename(self):
-        return self._soil_download.get_filename()
+    def get_tempfile(self):
+        return transient_file_store.get_tempfile(self.upload_id)
 
     def check_file(self, named_columns):
         """
         open a spreadsheet download ref just to test there are no errors opening it
 
         :raise ImporterError subtypes
         """
-        open_spreadsheet_download_ref(self._soil_download, named_columns)
+        tempfile = self.get_tempfile()
+        if not tempfile:
+            raise ImporterRefError('file not found in cache')
+        open_spreadsheet_download_ref(tempfile, named_columns)
 
     def get_spreadsheet(self, named_columns):
-        return get_spreadsheet(self.get_filename(), named_columns)
+        return get_spreadsheet(self.get_tempfile(), named_columns)
 
     def trigger_upload(self, domain, config):
         from corehq.apps.case_importer.tasks import bulk_import_async
         task = bulk_import_async.delay(config, domain, self.upload_id)
+        original_filename = transient_file_store.get_filename(self.upload_id)
+        with open(self.get_tempfile()) as f:
+            identifier = persistent_file_store.write_file(f, original_filename)
+
+        case_upload_file_meta = CaseUploadFileMeta.objects.get(identifier=identifier)
```

Comment:
Hey @snopoke after taking a closer look I saw the reason in your words and went with the approach of having write_file return a file meta object. For transient storage, it just returns it as a namedtuple; for persistent storage, it returns the saved `CaseUploadFileMeta` object.